py4DSTEM/process/preprocess/preprocess.py

```diff
@@ -149,5 +150,114 @@ def bin_data_real(datacube, bin_factor):
         datacube.R_Nx,datacube.R_Ny,datacube.Q_Nx,datacube.Q_Ny = datacube.data.shape
         return datacube
 
+def filter_bad_pixels(
+    datacube,
+    thresh,
+    ind_compare=1,
+    return_mask=False
+    ):
+    """
+    This function performs pixel filtering to remove hot or cold pixels. We first compute a moving median filter,
+    applied to the mean diffraction image. Pixels with intensity int_0 are compared to a locally ordered median
```

Comment:
Can you explain what `thresh` does in the docstring? It's unclear. Something more like "pixels are compared to their 21 neighbors, and those with intensity greater than the `ind_compare`th brightest neighbor by `thresh` are marked as "hot" and replaced with the value of the 8-neighbor median in each diffraction pattern".

Yes, writing that out made me feel insane. But I feel we should describe it precisely. 

Also, does this actually detect cold pixels? I think it is only doing ones that are stuck high.

Reply:
Doc string expanded, function renamed to specify hot pixels.